packages/wallets/ledger/src/adapter.ts

```diff
@@ -147,9 +147,15 @@ export class LedgerWalletAdapter extends BaseSignerWalletAdapter {
         try {
             try {
                 const transport = this._transport;
-                if (!transport) throw new WalletNotConnectedError();
+                const publicKey = this._publicKey;
+                if (!transport || !publicKey) throw new WalletNotConnectedError();
+
+                const appConfig = await getAppConfiguration(transport);
+                if (appConfig.version < '1.8.0') throw new WalletSignMessageError('Signing off-chain messages requires Solana Ledger App 1.8.0 or later');
```

Comment:
Comparing strings this way is going to be dicey. I think this probably needs to covert the major and minor version to a number, ignore the patch version, and then make sure it's `< 1.8` right?

Reply:
Addressed here: https://github.com/anza-xyz/wallet-adapter/pull/1094/commits/a7ac19194e7258ef22c9549ff6ba31f14168ff39